packages/desktop-client/src/components/filters/EditFilter.js

```diff
@@ -0,0 +1,267 @@
+import React, { useEffect, useRef, useReducer } from 'react';
+import { useParams, useHistory, useLocation } from 'react-router-dom';
```

Comment:
The build is failing because `useHistory` is no longer present in the version of react-router we updated to. You’ll need to swap its usage out for `useNavigate`. `history.push('/url')` is now `navigate('/url')`, and `history.goBack()` is now `navigate(-1)`.

Reply:
I ended up not using that with my new implementation. I think I've taken care of all the fixes for the new React version.